manifests/init.pp

```diff
@@ -146,6 +146,7 @@
   $cacheupdatefrequency                     = $zabbix::params::server_cacheupdatefrequency,
   $startdbsyncers                           = $zabbix::params::server_startdbsyncers,
   $historycachesize                         = $zabbix::params::server_historycachesize,
+  String $historyindexcachesize             = $zabbix::params::server_historyindexcachesize,
```

Comment:
This doesn't allow undef. So it needs to be at least this:
```puppet
   Optional[String[1]] $historyindexcachesize = $zabbix::params::server_historyindexcachesize,
```

Even better would it be to enforce the string content. I assume the following would work:

```puppet
Pattern[/^\d+[k|K|m|M]?$/] $historyindexcachesize = $zabbix::params::server_historyindexcachesize,
```

Please keep in mind that all `=` should be aligned. You probably need to update the alignment of all other `=` in this section.

Reply:
Added and adjusted the alignment accordingly.